### PR TITLE
Feature/upmcfhir 263 add missing comp fields

### PR DESCRIPTION
**Pull request by 5t33**
#### What does this PR do?

This PR creates the following mappings:
- DocumentationOf.serviceEvent.effectivetime -> composition.event.period
- DocumentationOf.serviceEvent.performer.assignedEntity -> event.detail.reference (along with a new practitioner)
- ~Composition.event.Code.text -> "Primary Care Physician"~

#### Related JIRA tickets:
[UMMCFHIR-263](https://jira.amida.com/browse/UPMCFHIR-263)

#### New JIRA tickets for clinical review required?
Probably?

### CODE REVIEW

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java [hunk 1]
```diff
@@ -897,6 +903,29 @@ public EntryResult tClinicalDocument2Bundle(ClinicalDocument cdaClinicalDocument
 			}
 		}
 
+		// documentationOf.serviceEvent.assignedEntity -> attester[mode = professional]
```
Comment (mmccall) [inline src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: So I don't think we want to make a service event assigned entity an attester. Quoting the spec, "The serviceEvent/performer represents the healthcare providers involved in the current or pertinent historical care of the patient. Preferably, the patient’s key healthcare providers would be listed, particularly their primary physician and any active consulting physicians, therapists, and counselors." I'd be more inclined to add it as a detail reference to any of the event itself, but that's my take.

Comment (mmccall) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: But yes, Bill's comments seem to indicate this makes more sense as an attester. However, we already have two attestors supported. I could see this being mapped to either, my personal feeling is that it should be detail. If we just tell them we did it that way I'd think it'd work, but I'd be curious to hear Afsin's opinion.

Comment (mmccall) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: We should ticket out supporting code on the event detail, as something we can evaluate when we look at real data. the spec is a little weird on it (the example differs from the content), and I imagine what we see from Epic/Cerner will vary substantially.

**Review by mmccall — COMMENTED** [review]
I can see this going either as attester or as event.detail. There is an HL7 to CDA mapping available on the FHIR site that actually neither. My personal inclination is to event.detail.

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java (hunk 1 above)
Comment (5t33) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: Ok. Would we at all indicate that the individual is a primary car provider?

Comment (mmccall) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: Ideally you would with the code element I think.

**Comment by 5t33** [comment]
@mmccall @au2 Good to go for review.

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java [hunk 2]
```diff
@@ -918,6 +922,31 @@ public EntryResult tClinicalDocument2Bundle(ClinicalDocument cdaClinicalDocument
...
+
+				// documentationOf.serviceEvent.assignedEntity -> composition.event.detail
+				if (cdaServiceEvent.getPerformers() != null && !cdaServiceEvent.getPerformers().isEmpty()) {
+					for (Performer1 performer : cdaServiceEvent.getPerformers()) {
+						EntityResult entityResult = tAssignedEntity2Practitioner(performer.getAssignedEntity(),
+								bundleInfo);
+						result.updateFrom(entityResult);
+						Reference reference = entityResult.getPractitionerReference();
+						event.addDetail(reference);
+						event.addCode(new CodeableConcept().setText("Primary Care Physician"));
```
Comment (mmccall) [inline src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: so I don't know that we want to set this to primary care physician. I think Bill suggested it, but I think it's incorrect. PCPR just means care provisioning, presumably by the doctor referenced herein, but we can't infer that this is actually their primary care physician. The code is optional, so I think we should remove it.

Comment (mmccall) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: this shouldn't be set here either. If you look at where the code is on the gold file, it should only be populated once per each composition. It actually comes in on the gold file as 'burn by fire' and a SNOMED code, but that doesn't map to the HL7 codes so no idea what to do with that information.

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java [hunk 3]
```diff
@@ -2699,6 +2728,14 @@ public EntityResult tPerformer22Practitioner(Performer2 cdaPerformer2, IBundleIn
 		return tAssignedEntity2Practitioner(cdaPerformer2.getAssignedEntity(), bundleInfo);
 	}
 
+	public EntityResult tPerformer12Practitioner(Performer1 cdaPerformer1, IBundleInfo bundleInfo) {
```
Comment (mmccall) [inline src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java:2809]: you don't actually call this, you call tAssignedEntity2Practitioner. Is it necessary?  If you did mean to use it, be aware that you could potentially be making empty practitioner objects in the bundle and should check for that.

**Review by mmccall — CHANGES_REQUESTED** [review]
2 comments.

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java (hunk 3 above)
Comment (5t33) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java:2809]: Well I'd rather do it that way just to remain consistent. What would checking for that look like? Something like this?
```
if (entityResult.hasPractitioner()) {
  result.updateFrom(entityResult);
  Reference reference = entityResult.getPractitionerReference();
  event.addDetail(reference);
}
```

Comment (5t33) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java:2809]: What might create an empty practitioner?

Comment (mmccall) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java:2809]: ^yes basically. I don't think you would actually create one, just saying the function can return one. It wouldn't hurt to add a check there.

Comment (5t33) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java:2809]: Would it not have an assignedEntity?

**Review by mmccall — APPROVED** [review]

**Pull request merged**